setup.py

```diff
@@ -1,5 +1,32 @@
+import sys
 import setuptools
 
+
+# Taken from Numba
+def _guard_python_version(max_python):
+    version_module = None
+    try:
+        from packaging import version as version_module
+    except ImportError:
+        try:
+            from setuptools._vendor.packaging import version as version_module
+        except ImportError:
+            pass
+
+    if version_module is None:
+        return
+
+    current_python = version_module.parse(".".join(map(str, sys.version_info[:3])))
+    max_python = version_module.parse(max_python)
+
+    if not current_python < max_python:
+        raise RuntimeError(
+            f"Cannot install on Python version {current_python} as Python {max_python}+ is not yet supported."
+        )
+
+
+_guard_python_version(max_python="3.13")
```

Comment:
@masonproffitt this is basically what Numba has in https://github.com/numba/numba/blob/147c641fb337c41029e9e8832d7cb40be700e413/setup.py and conforms to what @henryiii is suggesting as an alternative to capping the `requires-python` metadata.